internal/command/diff_inputs.go

```diff
@@ -0,0 +1,387 @@
+
+	if app1Digest == app2Digest {
+		if c.quiet && !c.csv {
+			stdout.Printf("the inputs for %s and %s match", args[0], args[1])
+		}
+		exitFunc(0)
+	}
+	if c.quiet && !c.csv {
+		stdout.Printf("the inputs for %s and %s differ", args[0], args[1])
+	}
+	exitFunc(2)
+}
+
+func getDiffInputArgDetails(repo *baur.Repository, args []string) []*diffInputArgDetails {
+	var results []*diffInputArgDetails
+
+	for _, arg := range args {
+		app, task, runID := parseDiffSpec(arg)
+		results = append(results, &diffInputArgDetails{arg: arg, appName: app, taskName: task, runID: runID})
+	}
+
+	var mustHaveTasks []string
+	for _, argDetails := range results {
+		if argDetails.runID == "" {
+			mustHaveTasks = append(mustHaveTasks, argDetails.arg)
+		}
+	}
+
+	if len(mustHaveTasks) > 0 {
+		tasks := mustArgToTasks(repo, mustHaveTasks)
+
+		for _, task := range tasks {
+			for _, argDetails := range results {
+				if argDetails.runID == "" && task.AppName == argDetails.appName && task.Name == argDetails.taskName {
+					argDetails.task = task
+				}
+			}
+		}
+
+		for _, argDetails := range results {
+			if argDetails.runID == "" && argDetails.task == nil {
+				exitOnErr(fmt.Errorf("task not found for %s", argDetails.arg))
+			}
+		}
+	}
+
+	return results
+}
+
+// parseDiffSpec splits an argument into the app, task and runID components.
+// It relies on the fact that the arguments have already been validated
+// in the cobra.Command
+func parseDiffSpec(s string) (app, task, runID string) {
+	if strings.Contains(s, ".") {
+		spl := strings.Split(s, ".")
+		app = spl[0]
+		task = spl[1]
+
+		if strings.Contains(task, "^") {
+			caretIndex := strings.Index(task, "^")
+			runID = subStr(task, caretIndex, len(task))
+			task = subStr(task, 0, caretIndex)
+		}
+
+		return app, task, runID
+	}
+
+	return "", "", s
+}
+
+func (c *diffInputsCmd) getTaskRunInputs(repo *baur.Repository, argDetails *diffInputArgDetails) (*baur.Inputs, *storage.TaskRunWithID, error) {
+	taskRun, err := getTaskRun(repo, argDetails)
+	if err != nil {
+		exitOnErr(err)
+	}
+
+	var inputs *baur.Inputs
+	if taskRun == nil {
+		inputResolver := baur.NewInputResolver()
+
+		inputFiles, err := inputResolver.Resolve(ctx, repo.Path, argDetails.task)
+		if err != nil {
+			exitOnErr(err)
+		}
+
+		inputs = baur.NewInputs(baur.InputAddStrIfNotEmpty(inputFiles, c.inputStr))
+	} else {
+		psql := mustNewCompatibleStorage(repo)
+		storageInputs, err := psql.Inputs(ctx, taskRun.ID)
+
+		if err != nil {
+			exitOnErr(err)
+		}
+
+		// Convert the inputs from the DB into baur.Input interface implementation
+		var baurInputs []baur.Input
+		for _, input := range storageInputs {
+			baurInputs = append(baurInputs, &storageInput{input})
+		}
+
+		inputs = baur.NewInputs(baur.InputAddStrIfNotEmpty(baurInputs, c.inputStr))
+	}
+
+	return inputs, taskRun, nil
+}
+
+func getTaskRun(repo *baur.Repository, argDetails *diffInputArgDetails) (*storage.TaskRunWithID, error) {
+	if argDetails.task != nil {
```

Comment:
instead of checking if task is nil here, only call the function when it should do something